docs/introduction.org

```diff
@@ -0,0 +1,106 @@
+* Introduction
+
+Nowadays, multiple languages have emerged to compete with the well-established C language.
+In fact, C is widely used in many places, learned by a lot of engineers, and used for many purposes.
+C is a powerful tool, /but with great power comes great responsibility/, in fact, memory leaks can easily occur by mistake.
+
+That's why new languages come to find a better way to handle errors and memory.
+The Rust language relies strongly on its borrow checker to avoid safe memory but it comes at a cost: increased code complexity
+
+In another case, the Zig language has a purpose to replace the C language.
+Zig wants to simplify the programming development experience with no hidden control flow and no macro to avoid complex mistakes.
+In comparison with the Rust language, Zig uses explicit memory allocation to prevent memory leaks.
+
+In the sections below, certain aspects of this programming language will be explored in greater depth.
+
+** What is not this documentation
+This is *not* a beginner guide for Zig, this documentation wants to go in-depth in certain parts of the language.
```

Comment:
I would not say that it is not a begginer guide but rather not a guide to learn zig as a first language